scripts/examples/python/provision.py

```diff
@@ -65,7 +65,7 @@ def parseProvisionSpec(txt):
     doc = xml.dom.minidom.parseString(txt)
     all = doc.getElementsByTagName("provision")
     if len(all) != 1:
-        raise "Expected to find exactly one <provision> element"
+        raise Exception("Expected to find exactly one <provision> element")
```

Comment:
```suggestion
        raise ValueError("Expected to find exactly one <provision> element")
```
raise a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError):
* Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as [IndexError](https://docs.python.org/3/library/exceptions.html#IndexError).

Reference 1: https://docs.python.org/3/library/exceptions.html
Reference 2: https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/broad-exception-raised.html
> Raising exceptions that are too generic force you to catch exceptions generically too. It will force you to use a naked ``except Exception:`` clause. You might then end up catching exceptions other than the ones you expect to catch. This can hide bugs or make it harder to debug programs when unrelated errors are hidden.

Examples:

Problematic code:
  ```py
def small_apple(apple, length):
    if len(apple) < length:
        raise Exception("Apple is too small!")  # [broad-exception-raised]
    print(f"{apple} is proper size.")
  ```
Correct code:
```py
def small_apple(apple, length):
    if len(apple) < length:
        raise ValueError("Apple is too small!")
    print(f"{apple} is proper size.")
```

Reply:
Will change the code to include specific exception instead of generic exception.

Reply:
Done ,included in commit bea2f7cd95615c90e9894efc23974c48301b4b85